perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -187,32 +187,53 @@ class SizeDistributionIterator(object):
     dist: dict. The distribution to draw sizes from.
   """
 
-  # IMPLEMENTATION LIMITATION: right now, this object only supports
-  # point distributions, which contain only a single size.
-
   def __init__(self, dist):
-    if len(dist) != 1:
-      raise NotImplementedError(
-          'General distributions are not implemented yet.')
-
-    size, _ = dist.iteritems().next()
-    self.size = size
+    # This class chooses random numbers by using random.random() to
+    # pick a float in [0.0, 100.0), then converting that into a choice
+    # from our distribution. The data structure is two lists, sizes
+    # and cutoffs. If the random number is in the range [cutoffs[i-1],
+    # cutoffs[i]), then the random choice is sizes[i]. Cutoffs has an
+    # implied starting element 0.0, and its last element must be
+    # 100.0.
+    self.sizes = []
+    self.cutoffs = []
+
+    total_percent = 0.0
+    for size, percent in dist.iteritems():
```

Comment:
@meteorfox thanks a lot for your review! I really appreciate you taking a look at my code.

I am happy to sort the sizes here. If this ever becomes a general-purpose distribution class, @klausw points out that sorted numbers are a much nicer interface than unsorted. Responding to your comment below, this change will also simplify the unit tests, since we will know exactly which `random()` values correspond to which sizes.